libs/libc/stdio/lib_stdio_cb.c

```diff
@@ -0,0 +1,90 @@
+/****************************************************************************
+ * libs/libc/stdio/lib_stdio_cb.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <sys/types.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "libc.h"
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: lib_fseek_cb
+ ****************************************************************************/
+
+off_t lib_fseek_cb(FAR void *cookie, off_t *offset, int whence)
+{
+  int fd = (int)(intptr_t)cookie;
+  return lseek(fd, *offset, whence);
+}
+
+/****************************************************************************
+ * Name: lib_fwrite_cb
+ ****************************************************************************/
+
+ssize_t lib_fwrite_cb(FAR void *cookie, FAR const char *buf, size_t size)
+{
+  int fd = (int)(intptr_t)cookie;
+  return _NX_WRITE(fd, buf, size);
+}
+
+/****************************************************************************
+ * Name: lib_fread_cb
+ ****************************************************************************/
+
+ssize_t lib_fread_cb(FAR void *cookie, FAR char *buf, size_t size)
+{
+  int fd = (int)(intptr_t)cookie;
+  return _NX_READ(fd, buf, size);
+}
+
+/****************************************************************************
+ * Name: lib_fclose_cb
+ ****************************************************************************/
+
+int lib_fclose_cb(FAR void *cookie)
+{
+  int fd = (int)(intptr_t)cookie;
```

Comment:
why check cookie here

Reply:
There should be a check whether we have valid fd before we call `close()`. This was previously in `lib_fclose.c` but I moved it here since in `lib_fclose.c` we dont know whether it is fd or cookie.

Reply:
close will do the check for us